src/BufferStream.jl

```diff
@@ -122,6 +122,7 @@ function readavailable(bs::BufferStream)
         bs.chunk_read_idx = 1
         return chunkview
     end
+    return r
```

Comment:
Why are you adding `r` here?

Reply:
For clarity, but can remove. I always get a bit confused with `return` in `do`.